vyper/semantics/validation/local.py

```diff
@@ -194,6 +194,12 @@ def __init__(
                     )
                 },
             )
+
+            # Add references to `self` as standalone address
+            self_references = fn_node.get_descendants(vy_ast.Name, {"id": "self"})
+            standalone_self = [n for n in self_references if not isinstance(n.get_ancestor(), vy_ast.Attribute)]
```

Comment:
slightly cleaner use of the `get_ancestor()` api would be `n.get_ancestor(vy_ast.Attribute) is not None`

Reply:
Update: Reverting the change because this will not catch `x.y(self)`.